workspaces/quay/plugins/quay/src/api/index.ts

```diff
@@ -55,28 +52,22 @@ export const quayApiRef = createApiRef<QuayApiV1>({
 
 export type Options = {
   discoveryApi: DiscoveryApi;
-  configApi: ConfigApi;
   identityApi: IdentityApi;
 };
 
 export class QuayApiClient implements QuayApiV1 {
   // @ts-ignore
   private readonly discoveryApi: DiscoveryApi;
 
-  private readonly configApi: ConfigApi;
-
   private readonly identityApi: IdentityApi;
 
   constructor(options: Options) {
     this.discoveryApi = options.discoveryApi;
-    this.configApi = options.configApi;
     this.identityApi = options.identityApi;
   }
 
   private async getBaseUrl() {
-    const proxyPath =
-      this.configApi.getOptionalString('quay.proxyPath') || DEFAULT_PROXY_PATH;
-    return `${await this.discoveryApi.getBaseUrl('proxy')}${proxyPath}`;
+    return `${await this.discoveryApi.getBaseUrl('quay')}`;
```

Comment:
With this change, configuring quay-backend becomes a required step for the quay plugin. The old proxy configuration must be removed and new config values should be added to the app-config.yaml. Existing users will need to make these config changes when the upgrade. @PatAKnight do you have any thoughts on this?

Reply:
Hmm, if we want to support the proxy and the new backend API we could do something like:

```typescript
  private async getBaseUrl() {
    const proxyPath = this.configApi.getOptionalString('quay.proxyPath');
    const baseUrl = await this.discoveryApi.getBaseUrl(proxyPath ? 'proxy' : 'quay');

    return proxyPath ? `${baseUrl}${proxyPath}` : baseUrl;
  }
```
:thinking:

Reply:
Patrick recommended that we make the new backend plugin optional. So I can do something similar to the code above and re-add any proxy-related code and documentation back.

Reply:
I added a potential solution to support both the Proxy and the new quay-backend. I tested both locally and did not find any issues. You can see the changes here: https://github.com/backstage/community-plugins/pull/2640/commits/41807e37a6430a8a59648d2f85be3a313dc72ec5